Add Japanese translation (ja_JP)

tsukurun opened the pull request:
## Summary
  - Add complete Japanese (ja_JP) translation for Jamulus client
  - Based on Korean (ko_KR) translation template
  - All 627 active strings translated

  ## Related Issue
  Closes #3567

  ## Test
  - Generated .qm file successfully with `lrelease`
  - No untranslated strings remaining

Commit:
Add Japanese translation (ja_JP)

This adds a complete Japanese translation for the Jamulus client application.
Based on Korean (ko_KR) translation template, all strings have been translated
to Japanese for better localization support.

Related Issue: #3567

Commit:
Add ja_JP to TRANSLATIONS list in Jamulus.pro

ann0see commented on src/translation/translation_ja_JP.ts:
```diff
@@ -0,0 +1,4341 @@
+    <message>
+        <location filename="../util.cpp" line="557"/>
+        <source>Simplified Chinese</source>
+        <translation>簡体字中国語</translation>
```
Is this valid? I don't understand Chinese or Japanese, but these signs look chinese to me.

ann0see commented on src/translation/translation_ja_JP.ts:
```diff
@@ -0,0 +1,4341 @@
+    </message>
+    <message>
+        <location filename="../clientsettingsdlg.cpp" line="123"/>
+        <source>Under the Windows operating system the ASIO driver (sound card) can be selected using %1. If the selected ASIO driver is not valid an error message is shown and the previous valid driver is selected. Under macOS the input and output hardware can be selected.</source>
```
We should really be consistent with macOS instead of MacOS throughout the code...

But that's not your problem ;-)

gilgongo commented on src/translation/translation_ja_JP.ts:
```diff
@@ -0,0 +1,4341 @@
+    </message>
+    <message>
+        <location filename="../clientdlg.cpp" line="178"/>
+        <source>The CPU of the client or server is at 100%.</source>
```
Yes (unless the code only indicates 100% and not, say, 99%? Or is it about load average, I get confused on that). 

BTW (and not for this PR of course) the English phrasing of a lot the help text in the app is fairly inconsistent and could do with being in what's sometimes called "note form" for brevity. For example here it would be better as simply "Client or server CPU overloaded" or even "Client/server CPU overload".

softins reviewed (commented):
I don't know Japanese, but I have built and run this code, and verified that there are no accelerator key conflicts in client or server, that I could see in Japanese mode.

This PR should also include an addition to `util.cpp` around line 560 to credit the translator for Japanese.

gilgongo commented on src/translation/translation_ja_JP.ts:
```diff
@@ -0,0 +1,4341 @@
+    <message>
+        <location filename="../audiomixerboard.cpp" line="254"/>
+        <source>Grp</source>
+        <translation>グループ</translation>
```
> I don't know Japanese, but from what Google shows, maybe 组 might work, and be shorter?

I can reveal that I have a degree in Japanese language :-) I'm pretty sure グループ would almost certainly be the preferred rendering in this context.

Commit:
Add Japanese translator credit to About dialog

Added tsukurun as the Japanese (ja_JP) translator in the credits section of util.cpp.

softins reviewed (approved):
Looks good to me

Commit:
Update Japanese translation: clarify "contributions" as "code contributions"

Changed "貢献" to "コード貢献" for clearer context as suggested by gilgongo.
- "貢献の詳細" → "コード貢献の詳細"
- "Github 貢献者リスト" → "Github コード貢献者リスト"
- "貢献者" → "コード貢献者"

tsukurun commented:
Thank you all for reviewing and merging this PR! 🎉

  Special thanks to @softins, @ignotus666, @gilgongo, and @ann0see for your helpful feedback and guidance throughout the review process.

  I'm excited to see Japanese translation included in Jamulus 3.12.0!

ann0see commented:
CHANGELOG: Translation: Added Japanese translation
